### DISCUSSION TITLE
Contradiction in T&C and AGPL License

### DISCUSSION THREAD

**Discussion by Grafcube** [nammayatri/nammayatri#969] [Q&A] [answered]
I found the app today and wanted to give it a try. It looks interesting since it's open source. I read through the T&C linked on the first page of the app where you enter your phone number and I noticed this.

> 5.1 You shall not (i) license, sublicense, sell, resell, transfer, assign, distribute or otherwise commercially exploit or make available to any third party the software or App in any way; (ii) modify or make derivative works based upon the software or App; (iii) create Internet “links” to the software or “frame” or “mirror” any App on any other server or wireless or internet-based device; (iv) reverse engineer or access the App in order to (a) design or build a competitive product or software, (b) design or build a product using similar ideas, features, functions or graphics of the software or App, or (c) copy, reproduce, record, or make available to the public any ideas, features, functions or graphics of the software or App, or (v) launch an automated program or script, including, but not limited to, web spiders, web crawlers, web robots, web ants, web indexers, bots, viruses or worms, or any program which may make multiple server requests per second, or unduly burdens or hinders the operation and/or performance of the software or App.

I see that the repo is licensed under the AGPL-3.0 and I don't see a CLA anywhere. I'm no lawyer, but this looks like a contradiction to me. An explanation would be appreciated. TIA.

**Comment by Prithivirajkota** [comment]
Any contribution to open source code is governed by AGPL-3.0 open source license and not by app's terms and conditions.

**Reply by Grafcube** [reply]
Okay, but that doesn't really answer the question. Is the T&C a template that wasn't reviewed before being used? It would be completely expected to see it in a proprietary app but not here. At the very least, it seems like the AGPL takes priority here but it's still very strange and a little bit concerning to see.

**Comment by dayanidhicse** [comment, marked as answer]
Hi @Grafcube, 

Thank you for sending across this query. We believe, as an person interested in looking at the open mobility platform, you have gotten concerned with the terms and conditions of the Namma Yatri App.

Now to clarify your query, we clarify that the open source code (as uploaded on the github under the open source license) is different from the Namma Yatri app itself. While the Namma Yatri app uses the open source code at its core, there are other proprietary components in the app which are proprietary to us. The terms and conditions in the app are intended to safeguard the app and the intellectual property in the app entirely.

Having said that, serious developers and contributors do get access to the open source code through the github portal itself. It is essential that the open source code is accessed from the github portal itself to that all contributions or forks can be tracked and such contributions further shared with our developer community. Copying the source code from the app itself is not allowed and is discouraged since we will not be able to track contributions and further developments.

All code downloads related to Namma Yatri as provided under github are not intended to be safeguarded by the Namma Yatri terms and conditions.
